packages/fx-core/src/component/debug.ts

```diff
@@ -487,67 +494,115 @@ export async function generateLocalDebugSettingsCommon(
           TasksNext.mergeTasks
         );
       } else {
-        const launchConfigurations = isM365
-          ? LaunchNext.generateM365Configurations(includeFrontend, includeBackend, includeBot)
-          : (await useNewTasks(inputs.projectPath))
-          ? LaunchNext.generateConfigurations(includeFrontend, includeBackend, includeBot)
-          : Launch.generateConfigurations(includeFrontend, includeBackend, includeBot);
-        const launchCompounds = isM365
-          ? LaunchNext.generateM365Compounds(includeFrontend, includeBackend, includeBot)
-          : (await useNewTasks(inputs.projectPath))
-          ? LaunchNext.generateCompounds(includeFrontend, includeBackend, includeBot)
-          : Launch.generateCompounds(includeFrontend, includeBackend, includeBot);
-
-        const tasks = isM365
-          ? TasksNext.generateM365Tasks(
-              includeFrontend,
-              includeBackend,
-              includeBot,
-              programmingLanguage
-            )
-          : (await useNewTasks(inputs.projectPath))
-          ? TasksNext.generateTasks(
-              includeFrontend,
-              includeBackend,
-              includeBot,
-              includeFuncHostedBot,
-              programmingLanguage
-            )
-          : Tasks.generateTasks(
-              includeFrontend,
-              includeBackend,
-              includeBot,
-              includeSimpleAuth,
-              programmingLanguage
-            );
-
-        //TODO: save files via context api
         await fs.ensureDir(`${inputs.projectPath}/.vscode/`);
         await updateJson(
-          `${inputs.projectPath}/.vscode/launch.json`,
-          {
-            version: "0.2.0",
-            configurations: launchConfigurations,
-            compounds: launchCompounds,
-          },
-          LaunchNext.mergeLaunches
+          `${inputs.projectPath}/.vscode/settings.json`,
+          Settings.generateSettings(includeBackend || includeFuncHostedBot, isSpfx),
+          Settings.mergeSettings
         );
 
-        await updateJson(
-          `${inputs.projectPath}/.vscode/tasks.json`,
-          {
-            version: "2.0.0",
-            tasks: tasks,
-          },
-          TasksNext.mergeTasks
-        );
+        if (await useTransparentTasks(inputs.projectPath)) {
```

Comment:
Just for more readable code (avoid soo many conditions...). Consider creating `generator` arrays with all possible functions.
Anyway, take this as further improvements.

``` typescript
private readonly taskGenerators = ...; // <version, <isM365, function>>
...
{
    const version = ...; // legacy, next, transparent
    const isM365 = ...; // true | false
    const tasks = await taskGenerators[version][isM365](args...);
}
...
```

Reply:
Good idea. Let's improve it afterwards.